### PR TITLE
Added Starwars Dashboard with pagination

### PR DESCRIPTION
**Pull request by xatuke**
Changes:
Added a Dashboard which shows starwars data and you can paginate throught it.

Screenshots:
star1

### [DEMO](https://streamable.com/dsfus)

### CODE REVIEW

File: src/components/sets/People.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    `;
+    return all;
+  }
+
+  useEffect(() => {
+    const fetchpeople = async () => {
+      setLoading(true);
+      await client.query({query: getAllPeople()})
+      .then(result => {
+      const data = result.data.allPersons;
```
Comment (divyanshu-rawat) [inline src/components/sets/People.js]: `const { data: { allPersons } } = result;`
`setPeople(allPersons);`

File: src/components/sets/Planets.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    }    
+    `;
+    return all;
+  }
+  useEffect(() => {
+    const fetchPlanets = async () => {
+      setLoading(true);
+      await client.query({query: getAllPlanets()})
+      .then(result => {
+      const data = result.data.allPlanets;
```
Comment (divyanshu-rawat) [inline src/components/sets/Planets.js]:
```
const { data: { allPlanets } } = result;
setPlanets(allPlanets);
```

File: src/components/sets/Planets.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      planet.diameter,
+      JSON.stringify(planet.climate),
+      planet.gravity,
+      JSON.stringify(planet.terrain),
+      planet.population,
+      planet.surfaceWater
+  ]}
+  ))
+
+  const paginate = (pageNumber) => setCurrentPage(pageNumber);
```
Comment (divyanshu-rawat) [inline src/components/sets/Planets.js]: you don't need `()` around `pageNumber`, here.
These are minor fixes its up-to you whether you want to do it or not. 👍

File: src/components/sets/Species.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      }
+    }       
+    `;
+    return all;
+  }
+  useEffect(() => {
+    const fetchSpecies = async () => {
+      await client.query({query: getAllSpecies()})
+      .then(result => {
+      const data = result.data.allSpecies;
```
Comment (divyanshu-rawat) [inline src/components/sets/Species.js]: Maybe you can use the same logic I mentioned in other cases here as well. 👍

File: src/components/sets/Starships.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    }           
+    `;
+    return all;
+  }
+  useEffect(() => {
+    const fetchStarships = async () => {
+      setLoading(true);
+      await client.query({query: getAllStarships()})
+      .then(result => {
+      const data = result.data.allStarships;
```
Comment (divyanshu-rawat) [inline src/components/sets/Starships.js]: Same holds true for here as well. 👍

File: src/components/sets/Vehicles.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    }             
+    `;
+    return all;
+  }
+  useEffect(() => {
+    const fetchvehicles = async () => {
+      setLoading(true);
+      await client.query({query: getAllVehicles()})
+      .then(result => {
+      const data = result.data.allVehicles;
```
Comment (divyanshu-rawat) [inline src/components/sets/Vehicles.js]: Same holds true for here as well. 👍

**Review by divyanshu-rawat — COMMENTED** [review]
I have suggested some changes to follow new syntax and enhance best practises feel free to make changes if you want 👍 .

**Comment by crowded-geek** [comment]
@divyanshu-rawat Thanks for all the suggestions, this really helps me as I am new to React so I get to know better practices :) Will fix itall now :smile:

File: src/components/HeaderTabs.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import { Tabs, Tab } from '@patternfly/react-core';
+import '@patternfly/react-core/dist/styles/base.css';
+import Films from './sets/Films'
+import People from './sets/People';
+import Planets from './sets/Planets';
+import Species from './sets/Species';
+import Starships from './sets/Starships';
+import Vehicles from './sets/Vehicles';
+
+class HeaderTabs extends React.Component {
```
[the remark this answers is not in the document]
Comment (crowded-geek) [reply src/components/HeaderTabs.js]: I couldn't achieve this, maybe because Tabs is a patternfly component and when used with hooks, it doesn't work.

**Comment by xatuke** [comment]
let me know, if I should raise another PR as I see you have merged and reverted this one. @anukritijha

**Pull request merged**